v3/lints/cabf_br/lint_sub_cert_aia_contains_internal_names.go

```diff
@@ -0,0 +1,77 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type subCertAIAInternalName struct{}
+
+/************************************************************************
+BRs: 7.1.2.10.3
+CA Certificate Authority Information Access
+This extension MAY be present. If present, it MUST NOT be marked critical, and it MUST contain the
+HTTP URL of the CA’s CRL service.
+
+id-ad-ocsp        A HTTP URL of the Issuing CA's OCSP responder.
+id-ad-caIssuers   A HTTP URL of the Issuing CA's Certificate.
+*************************************************************************/
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "w_sub_cert_aia_contains_internal_names",
+		Description:   "Subscriber certificates authorityInformationAccess extension should contain the HTTP URL of the issuing CA’s certificate, for public certificates this should not be an internal name",
+		Citation:      "BRs: 7.1.2.10.3",
+		Source:        lint.CABFBaselineRequirements,
+		EffectiveDate: util.CABEffectiveDate,
+		Lint:          NewSubCertAIAInternalName,
+	})
+}
+
+func NewSubCertAIAInternalName() lint.LintInterface {
+	return &subCertAIAInternalName{}
+}
+
+func (l *subCertAIAInternalName) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c)
+}
+
+func (l *subCertAIAInternalName) Execute(c *x509.Certificate) *lint.LintResult {
+	for _, u := range c.OCSPServer {
+		purl, err := url.Parse(u)
+		if err != nil {
+			return &lint.LintResult{Status: lint.Fatal}
```

Comment:
I believe that what we're doing here is saying that if the URL is parseable and has a valid TLD, then it is probably resolvable (which is more tolerable than _actually_ doing an DNS lookup).

So just a minor clerical thing that `lint.Fatal` is typically used to signal an internal error (although you can find discrepancies...) so this parsing failure would be an example of a `lint.Error`.

```suggestion
			return &lint.LintResult{Status: lint.Error}
```